logstash-core/src/main/java/org/logstash/config/ir/compiler/EventCondition.java

```diff
@@ -0,0 +1,758 @@
+            public boolean fulfilled(final JrubyEventExtLibrary.RubyEvent event) {
+                return value.compareTo(
+                    (IRubyObject) event.getEvent().getUnconvertedField(field)
+                ) < 0;
+            }
+        }
+
+        private static final class FieldEqualsString implements EventCondition {
+
+            private final FieldReference field;
+
+            private final RubyString value;
+
+            private FieldEqualsString(final String field, final String value) {
+                this.field = PathCache.cache(field);
+                this.value = RubyUtil.RUBY.newString(value);
+            }
+
+            @Override
+            public boolean fulfilled(final JrubyEventExtLibrary.RubyEvent event) {
+                final Object val = event.getEvent().getUnconvertedField(field);
+                return value.equals(val);
+            }
+        }
+
+        private static final class FieldEqualsLong implements EventCondition {
+
+            private final FieldReference field;
+
+            private final long value;
+
+            private FieldEqualsLong(final String field, final long value) {
+                this.field = PathCache.cache(field);
+                this.value = value;
+            }
+
+            @Override
+            public boolean fulfilled(final JrubyEventExtLibrary.RubyEvent event) {
+                final Object val = event.getEvent().getUnconvertedField(field);
+                return val instanceof RubyInteger && ((RubyInteger) val).getLongValue() == value;
+            }
+        }
+
+        private static final class FieldEqualsField implements EventCondition {
+
+            private final FieldReference one;
+
+            private final FieldReference other;
+
+            private FieldEqualsField(final FieldReference one, final FieldReference other) {
+                this.one = one;
+                this.other = other;
+            }
+
+            @Override
+            public boolean fulfilled(final JrubyEventExtLibrary.RubyEvent event) {
+                return event.getEvent().getUnconvertedField(one)
+                    .equals(event.getEvent().getUnconvertedField(other));
+            }
+        }
+
+        private static final class FieldMatches implements EventCondition {
+
+            private final FieldReference field;
+
+            private final RubyString regex;
+
+            private FieldMatches(final String field, final String regex) {
+                this.field = PathCache.cache(field);
+                this.regex = RubyUtil.RUBY.newString(regex);
+            }
+
+            @Override
+            public boolean fulfilled(final JrubyEventExtLibrary.RubyEvent event) {
+                final Object tomatch = event.getEvent().getUnconvertedField(field);
+                return tomatch instanceof RubyString &&
+                    !((RubyString) tomatch).match(RubyUtil.RUBY.getCurrentContext(), regex).isNil();
+            }
+        }
```

Comment:
@ph yea sadly there is a measurable gain in JDK regex because we don't have to actually capture, we just need to `find` and the Ruby API doesn't seem to offer this. Very small hit though + JDK regex come with some other problems potentially (either I have to convert RubysString to `String` before the matching or deal with potential encoding issues when using the `ByteList` in `RubyString` as a `CharBuffer` directly)